app/src/main/java/com/babylon/wallet/android/presentation/settings/appsettings/WalletPreferencesSettingsScreen.kt

```diff
@@ -0,0 +1,176 @@
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.designsystem.theme.RadixTheme
+import com.babylon.wallet.android.designsystem.theme.RadixWalletTheme
+import com.babylon.wallet.android.presentation.settings.SettingsItem
+import com.babylon.wallet.android.presentation.ui.composables.BasicPromptAlertDialog
+import com.babylon.wallet.android.presentation.ui.composables.DefaultSettingsItem
+import com.babylon.wallet.android.presentation.ui.composables.RadixCenteredTopAppBar
+import com.babylon.wallet.android.presentation.ui.composables.SwitchSettingsItem
+import kotlinx.collections.immutable.ImmutableSet
+
+@Composable
+fun WalletPreferencesScreen(
+    viewModel: WalletPreferencesViewModel,
+    onWalletPreferenceItemClick: (SettingsItem.WalletPreferencesSettingsItem) -> Unit,
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    WalletPreferencesContent(
+        modifier = modifier.fillMaxSize(),
+        walletPreferences = state.settings,
+        onWalletPreferenceItemClick = onWalletPreferenceItemClick,
+        onDeveloperModeToggled = viewModel::onDeveloperModeToggled,
+        onBackClick = onBackClick,
+        onCrashReportingToggled = viewModel::onCrashReportingToggled
+    )
+}
+
+@Composable
+private fun WalletPreferencesContent(
+    modifier: Modifier = Modifier,
+    walletPreferences: ImmutableSet<PreferencesUiItem>,
+    onWalletPreferenceItemClick: (SettingsItem.WalletPreferencesSettingsItem) -> Unit,
+    onDeveloperModeToggled: (Boolean) -> Unit,
+    onBackClick: () -> Unit,
+    onCrashReportingToggled: (Boolean) -> Unit,
+) {
+    var crashReportingPromptVisible by remember { mutableStateOf(false) }
+    if (crashReportingPromptVisible) {
+        BasicPromptAlertDialog(
+            finish = { accepted ->
+                if (accepted) {
+                    onCrashReportingToggled(true)
+                }
+                crashReportingPromptVisible = false
+            },
+            titleText = stringResource(id = R.string.appSettings_crashReporting_title),
+            messageText = stringResource(id = R.string.appSettings_crashReporting_subtitle)
+        )
+    }
+    Scaffold(
+        modifier = modifier,
+        topBar = {
+            RadixCenteredTopAppBar(
+                title = "Preferences",
```

Comment:
This can also be commented as reminder to replace with a strings res and add to crowdin: // TODO crowdin